### PR TITLE
foot: add module

### LINKED ISSUE #1547: Foot terminal configs
Hello. Please consider to add foot terminal configs into home manager config structure.

### PR DESCRIPTION
**Pull request by plabadens**
### Description

Add a module for `foot`, a "fast, lightweight and minimalistic Wayland terminal emulator". Fixes #1547.

Homepage: https://codeberg.org/dnkl/foot

### Checklist

- [X] Change is backwards compatible.

- [X] Code formatted with `./format`.

- [X] Code tested through `nix-shell --pure tests -A run.all`.

- [X] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [X] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/doc/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [X] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [X] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

File: modules/programs/foot.nix [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+          mouse = {
+            hide-when-typing = "yes";
+          };
+        }
+      '';
+    };
+  };
+
+  config = mkMerge [
+    (mkIf cfg.enable {
```
Comment (rycee) [inline modules/programs/foot.nix:50]: No need to use `mkMerge` here
```suggestion
  config = mkIf cfg.enable {
```

File: modules/programs/foot.nix [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      home.packages = [ cfg.package ];
+
+      xdg.configFile."foot/foot.ini" = mkIf (cfg.settings != { }) {
+        source = iniFormat.generate "foot.ini" cfg.settings;
+      };
+
+      systemd.user.services = (mkIf cfg.server.enable {
```
Comment (rycee) [inline modules/programs/foot.nix:57]: Unnecessary `(`:
```suggestion
      systemd.user.services = mkIf cfg.server.enable {
```

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -64,6 +64,8 @@
 
 /modules/programs/firefox.nix                         @rycee
 
+/modules/programs/foot.nix                            @plabadens
```
Comment (rycee) [inline .github/CODEOWNERS:67]:
```suggestion
/modules/programs/foot.nix                            @plabadens
/tests/modules/programs/foot                          @plabadens
```

File: tests/modules/programs/foot/empty-settings.nix [hunk 4]
```diff
@@ -0,0 +1,16 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+{
+  config = {
+    programs.foot.enable = true;
+
+    nixpkgs.overlays =
+      [ (self: super: { alacritty = pkgs.writeScriptBin "dummy-foot" ""; }) ];
```
Comment (rycee) [inline tests/modules/programs/foot/empty-settings.nix:10]:
```suggestion
      [ (self: super: { foot = pkgs.writeScriptBin "dummy-foot" ""; }) ];
```

**Comment by rycee** [comment]
Thanks for the contribution! I've added a few comments.

File: tests/modules/programs/foot/empty-settings.nix (hunk 4 above)
Comment (plabadens) [reply tests/modules/programs/foot/empty-settings.nix:10]: Yes as you can see my implementation is based on the alacritty module. 😅

**Comment by plabadens** [comment]
Thanks for the review! I've addressed the issues you highlighted.

File: modules/programs/foot.nix [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    };
+
+    server.enable = mkEnableOption "Foot terminal server";
+
+    settings = mkOption {
+      type = iniFormat.type;
+      default = { };
+      description = ''
+        Configuration written to
+        <filename>~/.config/foot/foot.ini</filename>. See <link
```
Comment (ambroisie) [inline modules/programs/foot.nix:28]:
```suggestion
        <filename>$XDG_CONFIG_HOME/foot/foot.ini</filename>. See <link

Comment (plabadens) [reply modules/programs/foot.nix:28]: Yes, that's more accurate, although it seems existing modules don't always do this.

**Pull request merged**